packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,472 @@
+            top: height - (arr.length - index) * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+    </div>
+  );
+};
+
+const calcRenderedHeight = (renderedItems: React.ReactElement[], stickyItemSize: number, itemSize: number) => {
+  let renderedHeight = 0;
+  renderedItems.forEach(item => {
+    renderedHeight = renderedHeight + (item.props.level === 1 ? stickyItemSize : itemSize);
+  });
+  return renderedHeight;
+};
+
+interface ReRenderStickyTreeProps {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  stickyItemSize: number;
+  itemSize: number;
+  height: number;
+  toplvlItemsId: string[];
+  onActiveItemsIdChange: (idsToRemove?: string[], idsToAdd?: string[]) => void;
+}
+
+interface ItemDataType {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  isNonSticky: (item: React.ReactElement<TreeItemProps>) => boolean;
+  onClick: (event: React.MouseEvent<HTMLElement>, itemProps: TreeItemProps) => void;
+  handleOnFocus: (event: React.FocusEvent<HTMLElement>) => void;
+  handleFocusParent: ComponentEventHandler<TreeItemProps>;
+  handleFocusFirstChild: ComponentEventHandler<TreeItemProps>;
+  handleOnSiblingsExpand: ComponentEventHandler<TreeItemProps>;
+}
+
+const ReRenderStickyTree = ({
+  renderedItems,
+  stickyItemSize,
+  itemSize,
+  height,
+  toplvlItemsId,
+  onActiveItemsIdChange,
+}: ReRenderStickyTreeProps) => {
+  const [stickyTopIds, setStickyTopIds] = React.useState<string[]>([toplvlItemsId[0]]);
+  const [stickyBottomIds, setStickyBottomIds] = React.useState<string[]>(toplvlItemsId.slice(1));
+
+  const passOnStickyTopIds = React.useCallback((newStickyTopIds: string[]) => {
+    setStickyTopIds([...newStickyTopIds]);
+  }, []);
+  const passOnStickyBottomIds = React.useCallback((newStickyBottomIds: string[]) => {
+    setStickyBottomIds([...newStickyBottomIds]);
+  }, []);
+
+  // when navigate with keyboard, use `itemIdTobeFocused` to remember the focused item id,
+  // then when DOM updates are finished, sticky tree knows which item to focus on
+  const [itemIdTobeFocused, setItemIdTobeFocused] = React.useState<string>(null);
+  const passItemIdTobeFocused = React.useCallback((newItemIdTobeFocused: string) => {
+    setItemIdTobeFocused(newItemIdTobeFocused);
+  }, []);
+
+  React.useEffect(() => {
+    // retain focus after sticky items gotten clicked and triggers DOM updates
+    if (itemIdTobeFocused === null) return;
+
+    const toFocusIndex = renderedItems.findIndex(item => item.props.id === itemIdTobeFocused);
+    const toFocusItem = renderedItems[toFocusIndex];
+    const toFocus = (toFocusItem?.props.contentRef as any)?.current;
+    if (toFocus) {
+      if (toFocusItem.props.level !== 1) ref.current.scrollToItem(toFocusIndex, 'center');
```

Comment:
```suggestion
    if (toFocus&& toFocusItem.props.level !== 1) ref.current.scrollToItem(toFocusIndex, 'center');
```

Reply:
I have to revert this. 
`if (toFocus)` actually contains 2 lines. I neglected this as well when I commit suggestion :P